src/geom/TileMatrix.js

```diff
@@ -0,0 +1,144 @@
+         * @param matrixWidth the number of tiles in the x direction
+         * @param matrixHeight the number of tiles in the y direction
+         * @param tileWidth the number of pixels or points in the x direction
+         * @param tileHeight the number of pixels or points in the y direction
+         * @constructor
+         */
+        var TileMatrix = function (sector, ordinal, matrixWidth, matrixHeight, tileWidth, tileHeight) {
+            if (!sector) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "missingSector"));
+            }
+
+            if (ordinal < 0) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "The specified oridinal is null or undefined."));
+            }
+
+            if (matrixWidth < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidWidth"));
+            }
+
+            if (matrixHeight < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidHeight"));
+            }
+
+            if (tileWidth < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidWidth"));
+            }
+
+            if (tileHeight < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidHeight"));
+            }
+
+            /**
+             * The geographic coverage area of this TileMatrix.
+             */
+            this.sector = sector;
+
+            /**
+             * The index position of this TileMatrix in the TileMatrixSet.
+             */
+            this.ordinal = ordinal;
+
+            /**
+             * The number of pixels or points of tiles in the horizontal or x direction.
+             */
+            this.tileWidth = tileWidth;
+
+            /**
+             * The number of pixels or points of tiles in the vertical or y direction.
+             */
+            this.tileHeight = tileHeight;
+
+            /**
+             * The number of tiles in the horizontal or x direction.
+             */
+            this.matrixWidth = matrixWidth;
+
+            /**
+             * The number of tiles in the vertical or y direction.
+             */
+            this.matrixHeight = matrixHeight;
+        };
+
+        /**
+         * Calculate the resolution of this TileMatrix in degrees per pixel
+         * @returns {number} degress per pixel resolution
+         */
+        TileMatrix.prototype.degreesPerPixel = function () {
```

Comment:
Could this be a property populated at construction time similar to Level.texelSize? It might make sense if the underlying properties (sector, tileHeight, etc.) aren't too dynamic.